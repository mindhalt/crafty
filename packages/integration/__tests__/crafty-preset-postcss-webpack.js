/* global describe, it, expect */

const path = require("path");
const rmfr = require("rmfr");
const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

it("Compiles CSS within webpack", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-postcss-webpack/compiles"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeTruthy();

  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myBundle.min.js")
  ).toMatchSnapshot();
});

it("Fails gracefully on broken markup", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-postcss-webpack/fails"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option)", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-postcss-webpack/extract-boolean"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeTruthy();
  expect(
    testUtils.exists(cwd, "dist/js/myBundle-default.min.css")
  ).toBeTruthy();
  expect(
    testUtils.exists(cwd, "dist/js/myBundle-default.min.css.map")
  ).toBeTruthy();

  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myBundle.min.js")
  ).toMatchSnapshot();
  expect(
    testUtils.readFile(cwd, "dist/js/myBundle-default.min.css")
  ).toMatchSnapshot();
});

it("Compiles CSS within webpack, extracts CSS ('extractCSS' string option)", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-postcss-webpack/extract-string"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle-string.min.css")).toBeTruthy();
  expect(
    testUtils.exists(cwd, "dist/js/myBundle-string.min.css.map")
  ).toBeTruthy();

  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myBundle.min.js")
  ).toMatchSnapshot();
  expect(
    testUtils.readFile(cwd, "dist/js/myBundle-string.min.css")
  ).toMatchSnapshot();
});

it("Compiles CSS within webpack, extracts CSS ('extractCSS' object option)", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-postcss-webpack/extract-object"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle-object.min.css")).toBeTruthy();
  expect(
    testUtils.exists(cwd, "dist/js/myBundle-object.min.css.map")
  ).toBeTruthy();

  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myBundle.min.js")
  ).toMatchSnapshot();
  expect(
    testUtils.readFile(cwd, "dist/js/myBundle-object.min.css")
  ).toMatchSnapshot();
});
