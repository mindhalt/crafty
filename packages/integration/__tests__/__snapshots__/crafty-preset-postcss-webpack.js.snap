// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles CSS within webpack 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/compiles/dist/css
           js: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                           status
  1 kB  0        multi ./js/app.js                built
  1 kB  1        ./css/style.scss                 built
  1 kB  2        postcss-loader!./css/style.scss  built
  1 kB  5        ./js/app.js + 1 modules          built

  size  name     asset                            status
  1 kB  default  myBundle.min.js                  emitted
  1 kB  default  myBundle.min.js.map              emitted

  (2 modules hidden)

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack 2`] = `
"(function(e,t){if(\\"object\\"===typeof exports&&\\"object\\"===typeof module)module.exports=t();else if(\\"function\\"===typeof define&&define.amd)define([],t);else{var n=t();for(var r in n)(\\"object\\"===typeof exports?exports:e)[r]=n[r]}})(window,(function(){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:false,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=true,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:true,get:r})},n.r=function(e){\\"undefined\\"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:true})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&\\"object\\"===typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,\\"default\\",{enumerable:true,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=0)}([function(e,t,n){e.exports=n(5)},function(e,t,n){var r=n(2);\\"string\\"===typeof r&&(r=[[e.i,r,\\"\\"]]);var o={};o.insert=\\"head\\",o.singleton=false;n(4)(r,o);r.locals&&(e.exports=r.locals)},function(e,t,n){t=e.exports=n(3)(false),t.push([e.i,\\"a{color:#00f}.Child{background:#000}.Parent{background:#fff}.Parent--before{color:#333}.Parent--after{color:#eee}.Button{color:orange}.Menu{width:800px}.MenuLink{color:red;background:#056ef0;width:200px}\\",\\"\\"])},function(e,t,n){\\"use strict\\";function r(e,t){var n=e[1]||\\"\\";var r=e[3];if(!r)return n;if(t&&\\"function\\"===typeof btoa){var a=o(r);var i=r.sources.map((function(e){return\\"/*# sourceURL=\\".concat(r.sourceRoot).concat(e,\\" */\\")}));return[n].concat(i).concat([a]).join(\\"\\\\n\\")}return[n].join(\\"\\\\n\\")}function o(e){var t=btoa(unescape(encodeURIComponent(JSON.stringify(e))));var n=\\"sourceMappingURL=data:application/json;charset=utf-8;base64,\\".concat(t);return\\"/*# \\".concat(n,\\" */\\")}e.exports=function(e){var t=[];return t.toString=function(){return this.map((function(t){var n=r(t,e);if(t[2])return\\"@media \\".concat(t[2],\\"{\\").concat(n,\\"}\\");return n})).join(\\"\\")},t.i=function(e,n){\\"string\\"===typeof e&&(e=[[null,e,\\"\\"]]);var r={};for(var o=0;o<this.length;o++){var a=this[o][0];null!=a&&(r[a]=true)}for(var i=0;i<e.length;i++){var u=e[i];null!=u[0]&&r[u[0]]||(n&&!u[2]?u[2]=n:n&&(u[2]=\\"(\\".concat(u[2],\\") and (\\").concat(n,\\")\\")),t.push(u))}},t}},function(e,t,n){\\"use strict\\";var r={};var o=function(){var e;return function(){return\\"undefined\\"===typeof e&&(e=Boolean(window&&document&&document.all&&!window.atob)),e}}();var a=function(){var e={};return function(t){if(\\"undefined\\"===typeof e[t]){var n=document.querySelector(t);if(window.HTMLIFrameElement&&n instanceof window.HTMLIFrameElement)try{n=n.contentDocument.head}catch(e){n=null}e[t]=n}return e[t]}}();function i(e,t){var n=[];var r={};for(var o=0;o<e.length;o++){var a=e[o];var i=t.base?a[0]+t.base:a[0];var u=a[1];var c=a[2];var f=a[3];var s={css:u,media:c,sourceMap:f};r[i]?r[i].parts.push(s):n.push(r[i]={id:i,parts:[s]})}return n}function u(e,t){for(var n=0;n<e.length;n++){var o=e[n];var a=r[o.id];var i=0;if(a){for(a.refs++;i<a.parts.length;i++)a.parts[i](o.parts[i]);for(;i<o.parts.length;i++)a.parts.push(h(o.parts[i],t))}else{var u=[];for(;i<o.parts.length;i++)u.push(h(o.parts[i],t));r[o.id]={id:o.id,refs:1,parts:u}}}}function c(e){var t=document.createElement(\\"style\\");if(\\"undefined\\"===typeof e.attributes.nonce){var r=n.nc;r&&(e.attributes.nonce=r)}if(Object.keys(e.attributes).forEach((function(n){t.setAttribute(n,e.attributes[n])})),\\"function\\"===typeof e.insert)e.insert(t);else{var o=a(e.insert||\\"head\\");if(!o)throw new Error(\\"Couldn't find a style target. This probably means that the value for the 'insert' parameter is invalid.\\");o.appendChild(t)}return t}function f(e){if(null===e.parentNode)return false;e.parentNode.removeChild(e)}var s=function(){var e=[];return function(t,n){return e[t]=n,e.filter(Boolean).join(\\"\\\\n\\")}}();function l(e,t,n,r){var o=n?\\"\\":r.css;if(e.styleSheet)e.styleSheet.cssText=s(t,o);else{var a=document.createTextNode(o);var i=e.childNodes;i[t]&&e.removeChild(i[t]),i.length?e.insertBefore(a,i[t]):e.appendChild(a)}}function d(e,t,n){var r=n.css;var o=n.media;var a=n.sourceMap;if(o&&e.setAttribute(\\"media\\",o),a&&btoa&&(r+=\\"\\\\n/*# sourceMappingURL=data:application/json;base64,\\".concat(btoa(unescape(encodeURIComponent(JSON.stringify(a)))),\\" */\\")),e.styleSheet)e.styleSheet.cssText=r;else{while(e.firstChild)e.removeChild(e.firstChild);e.appendChild(document.createTextNode(r))}}var p=null;var v=0;function h(e,t){var n;var r;var o;if(t.singleton){var a=v++;n=p||(p=c(t)),r=l.bind(null,n,a,false),o=l.bind(null,n,a,true)}else n=c(t),r=d.bind(null,n,t),o=function(){f(n)};return r(e),function(t){if(t){if(t.css===e.css&&t.media===e.media&&t.sourceMap===e.sourceMap)return;r(e=t)}else o()}}e.exports=function(e,t){t=t||{},t.attributes=\\"object\\"===typeof t.attributes?t.attributes:{},t.singleton||\\"boolean\\"===typeof t.singleton||(t.singleton=o());var n=i(e,t);return u(n,t),function(e){var o=[];for(var a=0;a<n.length;a++){var c=n[a];var f=r[c.id];f&&(f.refs--,o.push(f))}if(e){var s=i(e,t);u(s,t)}for(var l=0;l<o.length;l++){var d=o[l];if(0===d.refs){for(var p=0;p<d.parts.length;p++)d.parts[p]();delete r[d.id]}}}}},function(e,t,n){\\"use strict\\";function r(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}n.r(t);n(1);n.d(t,\\"default\\",(function(){return o}));var o=function e(){r(this,e)}}])}));
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option) 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/extract-boolean/dist/css
           js: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/extract-boolean/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                        status
  1 kB  0        multi ./js/app.js             built
  1 kB  1        ./css/style.scss              built
  1 kB  2        ./js/app.js + 1 modules       built

  size  name     asset                         status
  1 kB  default  myBundle-default.min.css      emitted
  1 kB  default  myBundle-default.min.css.map  emitted
  1 kB  default  myBundle.min.js               emitted
  1 kB  default  myBundle.min.js.map           emitted

  (1 module hidden)

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option) 2`] = `
"(function(e,t){if(\\"object\\"===typeof exports&&\\"object\\"===typeof module)module.exports=t();else if(\\"function\\"===typeof define&&define.amd)define([],t);else{var n=t();for(var r in n)(\\"object\\"===typeof exports?exports:e)[r]=n[r]}})(window,(function(){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:false,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=true,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:true,get:r})},n.r=function(e){\\"undefined\\"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:true})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&\\"object\\"===typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,\\"default\\",{enumerable:true,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=0)}([function(e,t,n){e.exports=n(2)},function(e,t,n){},function(e,t,n){\\"use strict\\";function r(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}n.r(t);n(1);n.d(t,\\"default\\",(function(){return o}));var o=function e(){r(this,e)}}])}));
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' boolean option) 3`] = `
"a{color:#00f}.Child{background:#000}.Parent{background:#fff}.Parent--before{color:#333}.Parent--after{color:#eee}.Button{color:orange}.Menu{width:800px}.MenuLink{color:red;background:#056ef0;width:200px}

/*# sourceMappingURL=myBundle-default.min.css.map*/"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' object option) 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/extract-object/dist/css
           js: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/extract-object/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                       status
  1 kB  0        multi ./js/app.js            built
  1 kB  1        ./css/style.scss             built
  1 kB  2        ./js/app.js + 1 modules      built

  size  name     asset                        status
  1 kB  default  myBundle-object.min.css      emitted
  1 kB  default  myBundle-object.min.css.map  emitted
  1 kB  default  myBundle.min.js              emitted
  1 kB  default  myBundle.min.js.map          emitted

  (1 module hidden)

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' object option) 2`] = `
"(function(e,t){if(\\"object\\"===typeof exports&&\\"object\\"===typeof module)module.exports=t();else if(\\"function\\"===typeof define&&define.amd)define([],t);else{var n=t();for(var r in n)(\\"object\\"===typeof exports?exports:e)[r]=n[r]}})(window,(function(){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:false,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=true,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:true,get:r})},n.r=function(e){\\"undefined\\"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:true})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&\\"object\\"===typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,\\"default\\",{enumerable:true,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=0)}([function(e,t,n){e.exports=n(2)},function(e,t,n){},function(e,t,n){\\"use strict\\";function r(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}n.r(t);n(1);n.d(t,\\"default\\",(function(){return o}));var o=function e(){r(this,e)}}])}));
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' object option) 3`] = `
"a{color:#00f}.Child{background:#000}.Parent{background:#fff}.Parent--before{color:#333}.Parent--after{color:#eee}.Button{color:orange}.Menu{width:800px}.MenuLink{color:red;background:#056ef0;width:200px}

/*# sourceMappingURL=myBundle-object.min.css.map*/"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' string option) 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/extract-string/dist/css
           js: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/extract-string/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                       status
  1 kB  0        multi ./js/app.js            built
  1 kB  1        ./css/style.scss             built
  1 kB  2        ./js/app.js + 1 modules      built

  size  name     asset                        status
  1 kB  default  myBundle-string.min.css      emitted
  1 kB  default  myBundle-string.min.css.map  emitted
  1 kB  default  myBundle.min.js              emitted
  1 kB  default  myBundle.min.js.map          emitted

  (1 module hidden)

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' string option) 2`] = `
"(function(e,t){if(\\"object\\"===typeof exports&&\\"object\\"===typeof module)module.exports=t();else if(\\"function\\"===typeof define&&define.amd)define([],t);else{var n=t();for(var r in n)(\\"object\\"===typeof exports?exports:e)[r]=n[r]}})(window,(function(){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:false,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=true,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:true,get:r})},n.r=function(e){\\"undefined\\"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:true})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&\\"object\\"===typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,\\"default\\",{enumerable:true,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=0)}([function(e,t,n){e.exports=n(2)},function(e,t,n){},function(e,t,n){\\"use strict\\";function r(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}n.r(t);n(1);n.d(t,\\"default\\",(function(){return o}));var o=function e(){r(this,e)}}])}));
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles CSS within webpack, extracts CSS ('extractCSS' string option) 3`] = `
"a{color:#00f}.Child{background:#000}.Parent{background:#fff}.Parent--before{color:#333}.Parent--after{color:#eee}.Button{color:orange}.Menu{width:800px}.MenuLink{color:red;background:#056ef0;width:200px}

/*# sourceMappingURL=myBundle-string.min.css.map*/"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            css: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/fails/dist/css
           js: __PATH__/packages/integration/fixtures/crafty-preset-postcss-webpack/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name  module                           status
  1 kB  0     multi ./js/app.js                built
  1 kB  1     ./css/style.scss                 built
  1 kB  2     postcss-loader!./css/style.scss  built failed ✖
  1 kB  4     ./js/app.js + 1 modules          built ✖

  (2 assets, 1 module hidden)

  Failed to compile.

./css/style.scss (__PATH__/node_modules/css-loader/dist/cjs.js??ref--4-1!__PATH__/node_modules/postcss-loader/src??ref--4-2!./css/style.scss)
Syntax error: Unknown word (10:13)

   8 |     @at-root {
   9 |         .Child
> 10 |             background: black;
     |             ^
  11 |         }
  12 |     }

  Δt ____ms ✖ 1 problem (1 error, 0 warnings)

[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Webpack compilation failed
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;
